pkg/auth/providers/ldap/ldap_provider.go

```diff
@@ -141,6 +145,46 @@ func (p *ldapProvider) GetPrincipal(principalID string, token v3.Token) (v3.Prin
 		return v3.Principal{}, err
 	}
 
+	if p.samlSearchProvider() {
+		lConn, err := ldap.Connect(config, caPool)
+		if err != nil {
+			return v3.Principal{}, nil
+		}
+		defer lConn.Close()
+
+		if scope == p.userScope {
+			userPrincipals, err := p.searchUser(externalID, config, lConn)
+			if err != nil {
+				return v3.Principal{}, err
+			}
+
+			if len(userPrincipals) != 1 {
+				return v3.Principal{}, fmt.Errorf("get principal for %s returned %d results",
+					externalID, len(userPrincipals))
+			}
+
+			principal := &userPrincipals[0]
+			if p.isThisUserMe(token.UserPrincipal, *principal) {
+				principal.Me = true
+			}
+
+			return *principal, nil
+		}
+
+		if scope == p.groupScope {
+			groupPrincipals, err := p.searchGroup(externalID, config, lConn)
```

Comment:
Same as above. You are doing a hard lookup, not a fuzzy search. Should probably have a more explicit query.